### PR TITLE
`azurerm_container_group` - Support updating managed identity

### LINKED ISSUE #24809: azurerm_container_group does not provision user assigned managed identity
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.1

### AzureRM Provider Version

3.90

### Affected Resource(s)/Data Source(s)

azurerm_container_group

### Terraform Configuration Files

```hcl
resource "azurerm_container_group" "aci" {
...
identity {
    type         = "SystemAssigned, UserAssigned"
    identity_ids = ["someresourceid"]
  }
...
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

The user assigned managed identity should be configured.

### Actual Behaviour

The user assigned managed identity is not added. Neither the type identity type is switched to "SystemAssigned, UserAssigned" and is stuck with "SystemAssigned":

 identity {
          ~ identity_ids = [
              + "someresourceid",
            ]
          ~ type         = "SystemAssigned" -> "SystemAssigned, UserAssigned"
            # (2 unchanged attributes hidden)
        }
}

After many applies, this is still reported by the plan.

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ning-kang**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR fixes Issue #24809 by partially implementing the update function of `azurerm_container_group`.
It allows update to the managed identity of container instance.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #24809

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/containers/container_group_resource.go [hunk 1]
```diff
@@ -840,13 +840,26 @@ func resourceContainerGroupUpdate(d *pluginsdk.ResourceData, meta interface{}) e
 		return err
 	}
 
-	t := d.Get("tags").(map[string]interface{})
+	existing, err := client.ContainerGroupsGet(ctx, *id)
+	if err != nil || existing.Model == nil {
+		return fmt.Errorf("reading Container Group %s: %v", id, err)
+	}
```
Comment (tombuildsstuff) [inline internal/services/containers/container_group_resource.go:846]: can we make this:

```suggestionggestion
	if err != nil {
		return fmt.Errorf("reading %s: %v", id, err)
	}
	if existing.Model == nil {
		return fmt.Errorf("reading %s: `model` was nil", id)
	}
```

**Review by tombuildsstuff — COMMENTED** [review]
hey @ning-kang 

Thanks for this PR - I've taken a look through and left one minor point about the error message for this one, but if we can fix that up then this otherwise LGTM 👍 

Thanks!

File: internal/services/containers/container_group_resource.go (hunk 1 above)
Comment (ning-kang) [reply internal/services/containers/container_group_resource.go]: Fixed. Thanks for picking this up.

**Review by stephybun — APPROVED** [review]
Thanks @ning-kang LGTM 👍

**Pull request merged**